open-api/rest-catalog-open-api.py

```diff
@@ -905,6 +973,28 @@ class CreateTableRequest(BaseModel):
     properties: Optional[Dict[str, str]] = None
 
 
+class PlanContext(BaseModel):
+    select: List[str] = Field(..., description='A list of the selected columns')
+    project: Optional[List[str]] = Field(None, description='A list of the projections')
+    filter: Optional[Expression] = None
+    options: Dict[str, str]
+    snapshot_id: Optional[int] = Field(None, alias='snapshot-id')
+    from_snapshot_id: Optional[int] = Field(None, alias='from-snapshot-id')
```

Comment:
in `IncrementalScan`, we have `fromSnapshot` API/semantics of `inclusive` or `exclusive`. do we need to replicate the same behavior here?

Reply:
Not sure what you mean can you elaborate more on this?

Reply:
meaning if the from snapshot is included or excluded in the increment scan of (from, to)

Reply:
I think @rdblue suggestion is to use a json list called `snapshot-range`,  instead of `fromSnapshot` and `toSnapshot`. From the description he suggested start as `exclusive` and end as `inclusive`

> snapshot-range: a JSON list containing exactly 2 int64 snapshot IDs (if snapshot-id is not present) representing the start (exclusive) and end (inclusive) snapshots.

However @stevenzwu I think i see your point since the Java api for incremental scan has both inclusive and exclusive https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/IncrementalScan.java#L34

```
  ThisT fromSnapshotExclusive(long fromSnapshotId);

  ThisT fromSnapshotInclusive(long fromSnapshotId);
```